yt/frontends/ytdata/utilities.py

```diff
@@ -235,4 +235,4 @@ def _yt_array_hdf5_attr(fh, attr, val):
     # This is raised if no HDF5 equivalent exists.
     # In that case, save its string representation.
     except TypeError:
-        fh.attrs[str(attr)] = repr(val)
```

Comment:
I believe this is technically a bug fix, but I don't think that it ever mattered before this PR: in a lot of types, `repr()` and `str()` will give identical results, because one of them typically defers to the other unless both magic methods `__repr__` and `__str__` are implemented. All that is to say, I don't think its worth backporting, but it _is_ necessary within the context of this PR, otherwise, the geometry attribute may be saved as _the string_ `" <Geometry.CARTESIAN: 'cartesian'>"`, which of course doesn't roundtrip very well.